app/controllers/users_controller.rb

```diff
@@ -19,4 +21,12 @@ def destroy_user
     user&.destroy!
     sign_out if user
   end
+
+  def ensure_in_setup
+    redirect_to root_url if !session[:user_confirmation_token] && two_factor_enabled
```

Comment:
Why are we checking for the user confirmation token in the session? What happens if we remove that here?

Reply:
I finally figured this out and posted some suggestions that may make this easier to follow in Slack. Those are take it or leave it though.